src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php

```diff
@@ -63,103 +68,111 @@ public function __construct(?array $mutatorPrefixes = null, ?array $accessorPref
-                        $types[] = $type;
-                        continue 2;
+            try {
+                $types[] = $this->stringTypeResolver->resolve((string) $tagDocNode->value->type, $typeContext, $backwardCompatible);
+            } catch (UnsupportedException) {
+                if ($backwardCompatible && 'void' === (string) $tagDocNode->value->type) {
+                    return Type::void();
+                }
+            } catch (LogicException $e) {
+                // handle "parent" type without existing parent
+                if ($backwardCompatible && 'parent' === (string) $tagDocNode->value->type) {
+                    return Type::object('parent');
                 }
 
-                $types[] = new Type(Type::BUILTIN_TYPE_OBJECT, $type->isNullable(), $resolvedClass, $type->isCollection(), $type->getCollectionKeyTypes(), $type->getCollectionValueTypes());
+                throw $e;
             }
         }
 
-        if (!isset($types[0])) {
+        if (null === $type = $types[0] ?? null) {
             return null;
         }
 
-        if (!\in_array($prefix, $this->arrayMutatorPrefixes, true)) {
-            return $types;
+        if (!\in_array($prefix, $this->arrayMutatorPrefixes)) {
+            return $type;
         }
 
-        return [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT), $types[0])];
+        return Type::list($type);
     }
 
-    public function getTypesFromConstructor(string $class, string $property): ?array
+    /**
+     * @deprecated since Symfony 7.1, use "getType" instead.
+     */
+    public function getTypes(string $class, string $property, array $context = []): ?array
+    {
+        trigger_deprecation('symfony/property-info', '7.1', 'The "%s()" method is deprecated, use "%s::getType()" instead.', __METHOD__, self::class);
+
+        return BackwardCompatibilityHelper::convertTypeToLegacyTypes($this->getType($class, $property, $context, true));
+    }
+
+    public function getTypeFromConstructor(string $class, string $property): ?Type
     {
+        // BC layer to allow unconsistent legacy type extraction
+        // Can be removed as soon as PhpStanExtractor::getTypesFromConstructor() is removed (8.0).
+        $backwardCompatible = \func_get_args()[2] ?? false;
+
         if (null === $tagDocNode = $this->getDocBlockFromConstructor($class, $property)) {
             return null;
         }
 
-        $types = [];
-        foreach ($this->phpStanTypeHelper->getTypes($tagDocNode, $this->nameScopeFactory->create($class)) as $type) {
-            $types[] = $type;
-        }
+        $typeContext = $this->typeContextFactory->createFromClassName($class);
 
-        if (!isset($types[0])) {
-            return null;
-        }
+        return $this->stringTypeResolver->resolve((string) $tagDocNode->type, $typeContext, $backwardCompatible);
+    }
+
+    /**
+     * @deprecated since Symfony 7.1, use "getTypeFromConstructor" instead.
```

Comment:
```suggestion
     * @deprecated since Symfony 7.1, use "getTypeFromConstructor" instead
```

(nitpick) we don't treat phpdoc annotations' descriptions as sentences as per https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/c1a9344ce4aea47b7926f1128027ffda9de029e8/doc/rules/phpdoc/phpdoc_annotation_without_dot.rst#L2